crates/cdk/src/mint_url.rs

```diff
@@ -65,7 +88,11 @@ where
 {
     fn from(url: S) -> Self {
         let url: String = url.into();
-        Self(url.trim_end_matches('/').to_string())
+        let formatted_url = Self::format_url(&url);
+        match formatted_url {
+            Ok(url) => Self(url),
+            Err(_) => Self::default(),
+        }
     }
 }
```

Comment:
I think we should just remove this `From` and move this logic into the `FromStr` below having this `from` doesn't add much as if a user has a type `Into<String>` they can just call `MintUrl::from_str(&their_type.to_string())?` this avoids hiding the error case here by returning an empty string.